src/main/java/com/google/sps/servlets/UserDataServlet.java

```diff
@@ -0,0 +1,113 @@
+package com.google.sps.servlets;
+
+import com.google.api.client.extensions.appengine.http.UrlFetchTransport;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
+import java.io.*;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Collections;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles forum data (questions, comments, and replies) */
+@WebServlet("/account")
+public class UserDataServlet extends HttpServlet {
+  private static final String ID_PROPERTY = "id";
+  private static final String NAME_PROPERTY = "name";
+  private static final String EMAIL_PROPERTY = "email";
+  private static final String LIKED_PROPERTY = "liked";
+  private static final String COMMENT_PROPERTY = "comments";
+  private static final String IMAGE_PROPERTY = "image";
+
+  private static final String DATASTORE_USER = "User";
+
+  private String userId = "";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/plain");
+    response.getWriter().println(userId);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String idTokenString = request.getParameter("idtoken");
+    UrlFetchTransport transport = new UrlFetchTransport();
+
+    GsonFactory gsonFactory = new GsonFactory();
+
+    GoogleIdTokenVerifier verifier =
+        new GoogleIdTokenVerifier.Builder(UrlFetchTransport.getDefaultInstance(), gsonFactory)
+            .setAudience(
+                Collections.singletonList(
+                    "757099697912-i6jll98mfgochdo2vgjcovf64pepjesc.apps.googleusercontent.com"))
+            .build();
+
+    GoogleIdToken idToken;
+    try {
+      idToken = verifier.verify(idTokenString);
+    } catch (GeneralSecurityException e) {
+      idToken = null;
+    }
+    if (idToken != null) {
+      Payload payload = idToken.getPayload();
+
+      userId = payload.getSubject();
+      String email = payload.getEmail();
+      String name = (String) payload.get("name");
```

Comment:
Use `NAME_PROPERTY` instead of `"name"`

Reply:
My issue with this is that its a different "property" than the NAME_PROPERTY, even though they have the same name. The "name", "picture" etc are fields that are inherently in payload, so I feel it would be misleading to use NAME_PROPERTY here.